ouroboros-network-protocols/src/Ouroboros/Network/Protocol/LocalStateQuery/Type.hs

```diff
@@ -23,14 +23,15 @@ import           Network.TypedProtocol.Core
 
 import           Ouroboros.Network.Util.ShowProxy (ShowProxy (..))
 
+data FootprintL = SmallL | LargeL | WholeL
 
 -- | The kind of the local state query protocol, and the types of
 -- the states in the protocol state machine.
 --
 -- It is parametrised over the type of block (for points), the type of queries
 -- and query results.
 --
-data LocalStateQuery block point (query :: Type -> Type) where
+data LocalStateQuery block point (query :: FootprintL -> Type -> Type) where
```

Comment:
Effectively we have 3 times more queries.  Alternatively we could just define them more explicitly in `ouroboros-consensus`, don't we?

Reply:
Gone!